clients/src/main/java/org/apache/kafka/clients/producer/ProducerRecord.java

```diff
@@ -38,11 +38,14 @@
  */
 public class ProducerRecord<K, V> {
 
+    public static final Long DEFAULT_EXPECTED_OFFSET = -1L;
```

Comment:
Is there a strong reason why we shouldn't use `null` as the default/no-offset-provided value instead?